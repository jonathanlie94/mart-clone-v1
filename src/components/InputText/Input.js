import styled from 'styled-components';
import theme from 'ui/theme';

export default styled.input.attrs({
  type: 'text',
})`
  font-family: Arial, sans-serif;
  border-radius: 4px;
  padding: 0.25rem;
  margin: 0;
  box-sizing: border-box;
  width: 100%;
  border-width: 2px;
  border-style: solid;
  border-color: ${props =>
    props.error ? theme.colors.yellow : theme.colors.grey};
  background-color: ${theme.colors.white};
  outline-color: ${theme.colors.blue};
  outline-width: 4px;
`;
