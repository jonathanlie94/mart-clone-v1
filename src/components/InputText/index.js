/*
  An input textarea component, used in our create topic page.
*/

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from 'ui/theme';
import { isEmptyString } from 'utils/helpers';
import Input from './Input';

const ErrorMessage = styled.div`
  color: ${theme.colors.red};
  margin-top: ${theme.margins.small};
`;

class InputText extends PureComponent {
  render() {
    return (
      <div >
        <Input
          className={this.props.className}
          onBlur={this.props.onBlur}
          onChange={e => this.props.onChange(e.target.value)}
          value={this.props.value}
        />
        {isEmptyString(this.props.error) ? false :
        <ErrorMessage>
          {this.props.error}
        </ErrorMessage>
        }
      </div>
    );
  }
}

InputText.propTypes = {
  className: PropTypes.string,
  error: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
};

InputText.defaultProps = {
  className: '',
  error: '',
  onBlur: () => {},
};

export default InputText;
