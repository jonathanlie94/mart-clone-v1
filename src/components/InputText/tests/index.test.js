import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import InputText from '../';

test('should handle change correctly', () => {
  let val = 'B';
  const onChange = (v) => {
    val = v;
  };
  const renderedComponent = mount(
    <InputText onChange={onChange} value={val} />
    );
  renderedComponent
      .find('input')
      .simulate('change', { target: { value: 'BA' } });

  expect(val).toEqual('BA');
});

test('should match previous snapshot', () => {
  const renderedComponent = mount(<InputText onChange={() => {}} value="val" />);
  const tree = toJSON(renderedComponent);
  expect(tree).toMatchStyledComponentsSnapshot();
});
