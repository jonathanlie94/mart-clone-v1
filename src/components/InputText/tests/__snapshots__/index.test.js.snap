// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
.jDMZsH {
  font-family: Arial, sans-serif;
  border-radius: 4px;
  padding: 0.25rem;
  margin: 0;
  box-sizing: border-box;
  width: 100%;
  border-width: 2px;
  border-style: solid;
  border-color: #C8C8C8;
  background-color: #FFF;
  outline-color: #00C3FF;
  outline-width: 4px;
}

<InputText
  className=""
  error=""
  onBlur={[Function]}
  onChange={[Function]}
  value="val"
>
  <div>
    <styled.input
      className=""
      onBlur={[Function]}
      onChange={[Function]}
      value="val"
    >
      <input
        className="sc-bdVaJa jDMZsH"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="val"
      />
    </styled.input>
  </div>
</InputText>
`;
